contracts/src/systems/combat/contracts.cairo

```diff
@@ -958,8 +965,12 @@ mod combat_systems {
                 };
             }
 
-            let mut destroyed_building_category = BuildingCategory::None;
+            // drain stamina
+            army_stamina.amount = 0;
```

Comment:
if you set the amount to 0 but don't change the last_refill_tick doesn't it mean that it could still be full ? maybe use the refill function first before doing any stamina changes ? also use sset method to keep consistency

Reply:
updated

Reply:
thank u ser